### PR TITLE
Feat/kube anti-affinity

### PR DESCRIPTION
**Pull request by radireddy**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-5524 and #MANAGER-5525
| License          | BSD 3-Clause

## Description

Added anti-affinity support while creating node pool and kube cluster. Refer to jira MANAGER-5524 for more details.

Kube antiaffinity(MANAGER-5524) and monthly billing changes (MANAGER-5525, made by @JeremyDec ) are merged into this PR.

### :flags: Translations

- [x] TRANS-29241

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/kubernetes/add/add.controller.js [hunk 1]
```diff
@@ -128,5 +132,8 @@ export default class {
 
   onNodePoolSubmit() {
     this.displaySelectedFlavor = true;
+    if (this.cluster.nodePool.nodeCount > 5) {
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/add/add.controller.js:135]: 5 could be stored as a constant

File: packages/manager/modules/pci/src/projects/project/kubernetes/add/add.html [hunk 2]
```diff
@@ -60,12 +60,14 @@ <h1 data-translate="kubernetes_add"></h1>
     </oui-step-form>
 
     <oui-step-form
-        data-header="{{:: 'kube_common_billing_type_title' | translate }}"
+        data-header="{{:: 'kubernetes_add_billing_anti_affinite_title' | translate }}"
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/add/add.html:63]:
```suggestion
        data-header="{{:: 'kubernetes_add_billing_anti_affinity_title' | translate }}"
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/components/anti-affinity/anti-affinity.controller.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import has from 'lodash/has';
+import omit from 'lodash/omit';
+
+export default class NodePoolAntiAffinityCtrl {
+  onAntiAffinityChange(isAntiAffinity) {
+    if (isAntiAffinity) {
+      this.nodePool.maxNodes = 5;
+    } else if (has(this.nodePool, 'maxNodes')) {
+      omit(this.nodePool, ['maxNodes']);
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/components/anti-affinity/anti-affinity.controller.js:9]: No need to set a new value ? Im' not sure `omit` mutates the object (https://lodash.com/docs/4.17.15#omit)

File: packages/manager/modules/pci/src/projects/project/kubernetes/components/anti-affinity/anti-affinity.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<oui-checkbox
+    data-model="$ctrl.nodePool.antiAffinity"
+    disabled="$ctrl.nodePool.nodeCount > 5"
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/components/anti-affinity/anti-affinity.html:3]: The `5` value is repeated quite a few times, a constant or exposing it through a method like `getMaxNodeCount()` would be better to maintain

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/add/add.html [hunk 5]
```diff
@@ -50,7 +50,7 @@ <h1 data-translate="kube_common_create_node_pool"></h1>
     </oui-step-form>
 
     <oui-step-form
-        data-header="{{:: 'kube_common_billing_type_title' | translate }}"
+        data-header="{{:: 'kube_add_billing_anti_affinite_title' | translate }}"
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/add/add.html:53]:
```suggestion
        data-header="{{:: 'kube_add_billing_anti_affinity_title' | translate }}"
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/node-pool.html [hunk 6]
```diff
@@ -70,6 +70,25 @@
...
+            data-title=":: 'kube_node_pool_anti_affinity' | translate"
+            data-property="antiAffinity"
+            data-type="boolean"
+            data-sortable
+        >
+            <div
+                data-ng-if="$row.antiAffinity"
+                class="d-flex justify-content-center"
+            >
+                <i class="oui-icon oui-icon-check" aria-hidden="true"></i>
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/node-pool.html:83]:
```suggestion
                <span class="oui-icon oui-icon-check" aria-hidden="true"></span>
```
Also missing a `span` to describe the value

**Review by JeremyDec — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/node-pool.html [hunk 7]
```diff
@@ -70,6 +70,25 @@
...
+            data-title=":: 'kube_node_pool_anti_affinity' | translate"
+            data-property="antiAffinity"
+            data-type="boolean"
+            data-sortable
+        >
+            <div
+                data-ng-if="$row.antiAffinity"
+                class="d-flex justify-content-center"
+            >
+                <span class="oui-icon oui-icon-check" aria-hidden="true"></span>
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/node-pool.html:83]: A description should be added to describe the icons. We can use `sr-only` class if we only want it diplayed for screen readers

Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/node-pool.html:83]: A description should be added to describe the icons. We can use `sr-only` class if we only want it diplayed for screen readers

**Review by marie-j — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/nodes/add/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -25,5 +25,6 @@
   "kube_nodes_add_same_flavor_message": "Vous pouvez uniquement choisir ce type d'instance car le pool de nœuds ne peut être composé que de nœuds du même type d'instance.",
   "kube_nodes_add_select_nodes": "Sélectionnez le nombre de nouveaux nœuds à ajouter",
   "kube_nodes_add_instance_type": "Type d'instance",
-  "kube_nodes_add_adding_message": "Redimensionnement du pool"
+  "kube_nodes_add_adding_message": "Redimensionnement du pool",
+  "kube_nodes_add_anti_affinity_active_info": "L'anti-affinité est activée : le pool de nœuds ne pourra pas contenir plus de {{ maxNodes }} noeuds"
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/node-pool/nodes/add/translations/Messages_fr_FR.json:29]:
```suggestion
  "kube_nodes_add_anti_affinity_active_info": "L'anti-affinité est activée : le pool de nœuds ne pourra pas contenir plus de {{ maxNodes }} nœuds"
```

**Review by JeremyDec — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build#16481.0 ✘  
* 
  * Build ✘

**Review by antleblanc — CHANGES_REQUESTED** [review]
@radireddy Could you please signing following commit: 28e2d93f1e7e1ca1733b5d20d43d4e727249baa8

**Review by antleblanc — APPROVED** [review]

**Pull request merged**